force-app/main/default/lwc/programSettings/programSettings.js

```diff
@@ -0,0 +1,123 @@
+        autoEnrollmentMappingsTable: {
+            autoEnrollmentMappingsTitle: autoEnrollmentMappingsTitle,
+            autoEnrollmentMappingsDescription: autoEnrollmentMappingsDescription,
+            accountRecordTypeColumn: stgColAccountRecordType,
+            autoEnrollmentStatusColumn: stgColAutoEnrollmentStatus,
+            autoEnrollmentRoleColumn: stgColAutoEnrollmentRole,
+            editAction: stgBtnEdit,
+            deleteAction: stgBtnDelete,
+        },
+    };
+
+    inputAttributeReference = {};
+
+    get affordancesDisabled() {
+        if (!this.isEditMode || this.affordancesDisabledToggle === true) {
+            return true;
+        }
+        return undefined;
+    }
+
+    @wire(getProgramSettingsVModel)
+    programSettingsVModelWire(result) {
+        this.programSettingsVModelWireResult = result;
+
+        if (result.data) {
+            this.programSettingsVModel = result.data;
+        } else if (result.error) {
+            //console.log("error retrieving preferredContactInfoSettingsVModel");
+        }
+    }
+
+    get autoEnrollmentMappingsTableColumns() {
+        return [
+            {
+                label: this.labelReference.autoEnrollmentMappingsTable.accountRecordTypeColumn,
+                fieldName: "accountRecordTypeLabel",
+            },
+            {
+                label: this.labelReference.autoEnrollmentMappingsTable.autoEnrollmentStatusColumn,
+                fieldName: "autoProgramEnrollmentStatus",
+            },
+            {
+                label: this.labelReference.autoEnrollmentMappingsTable.autoEnrollmentRoleColumn,
+                fieldName: "autoProgramEnrollmentRole",
+            },
+            {
+                type: "action",
+                typeAttributes: {
+                    rowActions: [
+                        { label: this.labelReference.autoEnrollmentMappingsTable.editAction, name: "edit" },
+                        { label: this.labelReference.autoEnrollmentMappingsTable.deleteAction, name: "delete" },
+                    ],
+                },
+            },
+        ];
+    }
+
+    handleSettingsEditModeChange(event) {
+        this.isEditMode = !event.detail;
+        this.affordancesDisabledToggle = event.detail;
+    }
+
+    handleSettingsSaving(event) {
+        this.affordancesDisabledToggle = true;
+        // TODO: perform client side validation
+
+        // if validation fails, call this.handleValidationFailure()
+        //this.template.querySelector("c-settings-save-canvas").handleValidationFailure();
+
+        // else, update hierarchy settings
+        this.template.querySelector("c-settings-save-canvas").updateHierarchySettings();
+    }
+
+    handleSettingsSaveCompleted(event) {
+        this.affordancesDisabledToggle = false;
+        this.refreshAllApex();
+    }
+
+    handleSettingsSaveCancel(event) {
+        this.refreshAllApex();
+    }
+
+    refreshAllApex() {
+        /*Promise.all([
```

Comment:
Please add the view refresh to this as this is a necessary component for "viewing"

Reply:
Done!